1.preprocess_data/preprocess_training_data.py

```diff
@@ -0,0 +1,154 @@
+        background_timelapse = baseflour
+    )
+    
+    #convert corrected images to numpy array, normalize, and convert to uint8
+    brightfield_images_corrected = np.array(brightfield_images_corrected_original)
+    brightfield_images_corrected[brightfield_images_corrected<0] = 0 #make negatives 0
+    brightfield_images_corrected = brightfield_images_corrected / np.max(brightfield_images_corrected) # normalize the data to 0 - 1
+    brightfield_images_corrected = 255 * brightfield_images_corrected # Now scale by 255
+    corrected_movie = brightfield_images_corrected.astype(np.uint8)
+    
+    return corrected_movie
+
+
+# ### Preprocess all training movies and save frames that have labeled data
+
+# In[5]:
+
+
+def preprocess_training_movies(ij, training_frames_path, downloads_dir, save_dir):
+    training_locations = pd.read_csv(training_frames_path, sep="\t", dtype=object)
+    
+    #download each well from IDR, if it is available on IDR
+    for index, row in training_locations.iterrows():
+        plate = row["Plate"]
+        well = row["Well"]
+        frame = row["Frame"]
+        print(f"\nPreprocessing {plate} {well} {frame}")
+        try:
+            save_path = pathlib.Path(f"{save_dir}{plate}/{well}/")
+            if not save_path.exists():
+                #determine what frames in the movie are labeled
+                plate_well_frames = training_locations[(training_locations["Plate"]==row["Plate"]) & (training_locations["Well"]==row["Well"])]
+                frame_nums = plate_well_frames["Frame"].tolist()
+                print(f"Labeled frames: {frame_nums}")
+                
+                original_movie = load_training_movie_data(ij, downloads_dir, row["Plate"], row["Well"])
+                corrected_movie = pybasic_illumination_correction(original_movie)
+                
+                #save all frames that have already had illumination correction
+                for frame_num in frame_nums:
+                    try:
+                        labeled_frame = corrected_movie[int(frame_num)-1]
+                        frame_dir_save_path = save_path.joinpath(f"{frame_num}")
+                        frame_dir_save_path.mkdir(parents=True, exist_ok=True)
+                        frame_save_path = frame_dir_save_path.joinpath(f"{plate}_{well}_{frame_num}.tif")
+                        skimage.io.imsave(frame_save_path, labeled_frame)
+                    except Exception as e:
+                        print(e)
+            else:
+                print("Movie has already been preprocessed!")
+        except Exception as e:
+            print(e)
+    
+
+ij = imagej.init('Fiji.app')
+#imagej init sets directory to /Fiji.app so have to go back a directory :/
+os.chdir("..")
+
+training_frames_path = "training_frames.tsv"
+downloads_dir = "../0.download_data/labeled_movies_ch5/"
+save_dir = "labeled_frames_preprocessed/"
+
+test = preprocess_training_movies(ij, training_frames_path, downloads_dir, save_dir)
```

Comment:
```suggestion
preprocess_training_movies(ij, training_frames_path, downloads_dir, save_dir)
```

if the function doesn't output anything, no need to set a return variable

Reply:
Added to https://github.com/WayScience/mitocheck_data/pull/2/commits/bc2ddcf3cc589d0e903776a37990664cd00b2a50.
Changes included in Jupyter Notebook.